### PR TITLE
[Intersmash] - Adding a presubmit job variant that runs against productized deliverables

### PR DESCRIPTION
**Pull request by fabiobrz**
This will validate changes by running the Intersmash test suite against productized deliverables, but it won't be strictly required in order to merge Intersmash PRs.

### CODE REVIEW

**Comment by fabiobrz** [comment]
> @fabiobrz: The following test **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:
> Test name 	Commit 	Details 	Required 	Rerun command
> ci/rehearse/Intersmash/intersmash/main/prod-intersmash-e2e-prod 	[daab702](https://github.com/openshift/release/commit/daab702ee9952894f48fbe12155bc96f38020fb3) 	[link](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/43276/rehearse-43276-pull-ci-Intersmash-intersmash-main-prod-intersmash-e2e-prod/1701704057590124544) 	unknown 	`/pj-rehearse pull-ci-Intersmash-intersmash-main-prod-intersmash-e2e-prod`
> 
> [Full PR test history](https://prow.ci.openshift.org/pr-history?org=openshift&repo=release&pr=43276). [Your PR dashboard](https://prow.ci.openshift.org/pr?query=is:pr+state:open+author:fabiobrz).

The rehearsal job outcome is fine. The error is due to Kafka misconfiguration on the Intersmash side.
@tommaso-borgato , could you please have a look to the changes and `/lgtm` once you think they're acceptable?

File: ci-operator/config/Intersmash/intersmash/Intersmash-intersmash-main__prod.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    product: ocp
+    timeout: 1h0m0s
+    version: "4.12"
+  steps:
+    test:
+    - as: intersmash-prod-build
+      commands: |
+        ls -la .ci/openshift-ci/build-root/
+        /bin/bash .ci/openshift-ci/build-root/e2e-test-prod.sh
+      credentials:
```
Comment (tommaso-borgato) [inline ci-operator/config/Intersmash/intersmash/Intersmash-intersmash-main__prod.yaml:26]: where do this credentials are taken from?

File: ci-operator/config/Intersmash/intersmash/Intersmash-intersmash-main__prod.yaml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+      from: src
+      resources:
+        requests:
+          cpu: 100m
+    workflow: generic-claim
+zz_generated_metadata:
+  branch: main
+  org: Intersmash
+  repo: intersmash
+  variant: prod
```
Comment (tommaso-borgato) [inline ci-operator/config/Intersmash/intersmash/Intersmash-intersmash-main__prod.yaml:39]: what is this new `variant: prod` attribute?

File: ci-operator/config/Intersmash/intersmash/Intersmash-intersmash-main__prod.yaml (hunk 1 above)
Comment (fabiobrz) [reply ci-operator/config/Intersmash/intersmash/Intersmash-intersmash-main__prod.yaml:26]: This is similar to what we've done for the periodic job, see https://github.com/openshift/release/blob/master/ci-operator/config/Intersmash/intersmash/Intersmash-intersmash-main__periodic.yaml#L27-L30

The process for storing credentials in the vault used by the OpenShift CI infra is documented [here](https://docs.ci.openshift.org/docs/how-tos/adding-a-new-secret-to-ci/).
We've done this and consequently mount the volume that will contain such secrets.

File: ci-operator/config/Intersmash/intersmash/Intersmash-intersmash-main__prod.yaml (hunk 2 above)
Comment (fabiobrz) [reply ci-operator/config/Intersmash/intersmash/Intersmash-intersmash-main__prod.yaml:39]: Variants allow for having more than one job, we need this for having another presubmit job, that will test against released product deliverables, instead then against community ones.

**Pull request merged**